File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-confirmation-processes-5-4/ConfirmationStatus.cpp
// ConfirmationStatus.cpp 
#include "ConfirmationStatus.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ConfirmationStatus::ConfirmationStatus(TiXmlNode* xmlNode)
: ResponseMessage(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //statusNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* statusNode = xmlNode->FirstChildElement("status");

   if(statusNode){statusIsNull_ = false;}
   else{statusIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- statusNode , address : " << statusNode << std::endl;
   #endif
   if(statusNode)
   {
      if(statusNode->Attribute("href") || statusNode->Attribute("id"))
      {
          if(statusNode->Attribute("id"))
          {
             statusIDRef_ = statusNode->Attribute("id");
             status_ = boost::shared_ptr<EventStatus>(new EventStatus(statusNode));
             status_->setID(statusIDRef_);
             IDManager::instance().setID(statusIDRef_,status_);
          }
          else if(statusNode->Attribute("href")) { statusIDRef_ = statusNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { status_ = boost::shared_ptr<EventStatus>(new EventStatus(statusNode));}
   }

   //assertedEventNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* assertedEventNode = xmlNode->FirstChildElement("assertedEvent");

   if(assertedEventNode){assertedEventIsNull_ = false;}
   else{assertedEventIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- assertedEventNode , address : " << assertedEventNode << std::endl;
   #endif
   if(assertedEventNode)
   {
      if(assertedEventNode->Attribute("href") || assertedEventNode->Attribute("id"))
      {
          if(assertedEventNode->Attribute("id"))
          {
             assertedEventIDRef_ = assertedEventNode->Attribute("id");
             assertedEvent_ = boost::shared_ptr<EventsChoice>(new EventsChoice(assertedEventNode));
             assertedEvent_->setID(assertedEventIDRef_);
             IDManager::instance().setID(assertedEventIDRef_,assertedEvent_);
          }
          else if(assertedEventNode->Attribute("href")) { assertedEventIDRef_ = assertedEventNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { assertedEvent_ = boost::shared_ptr<EventsChoice>(new EventsChoice(assertedEventNode));}
   }

   //proposedMatchNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* proposedMatchNode = xmlNode->FirstChildElement("proposedMatch");

   if(proposedMatchNode){proposedMatchIsNull_ = false;}
   else{proposedMatchIsNull_ = true;}

   if(proposedMatchNode)
   {
      for(proposedMatchNode; proposedMatchNode; proposedMatchNode = proposedMatchNode->NextSiblingElement("proposedMatch")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- proposedMatchNode , address : " << proposedMatchNode << std::endl;
          #endif
          if(proposedMatchNode->Attribute("href") || proposedMatchNode->Attribute("id"))
          {
              if(proposedMatchNode->Attribute("id"))
              {
                  proposedMatchIDRef_ = proposedMatchNode->Attribute("id");
                  proposedMatch_.push_back(boost::shared_ptr<EventProposedMatch>(new EventProposedMatch(proposedMatchNode)));
                  proposedMatch_.back()->setID(proposedMatchIDRef_);
                  IDManager::instance().setID(proposedMatchIDRef_, proposedMatch_.back());
              }
              else if(proposedMatchNode->Attribute("href")) { proposedMatchIDRef_ = proposedMatchNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { proposedMatch_.push_back(boost::shared_ptr<EventProposedMatch>(new EventProposedMatch(proposedMatchNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- proposedMatchNode , address : " << proposedMatchNode << std::endl;
       #endif
   }

   //allegedEventNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* allegedEventNode = xmlNode->FirstChildElement("allegedEvent");

   if(allegedEventNode){allegedEventIsNull_ = false;}
   else{allegedEventIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- allegedEventNode , address : " << allegedEventNode << std::endl;
   #endif
   if(allegedEventNode)
   {
      if(allegedEventNode->Attribute("href") || allegedEventNode->Attribute("id"))
      {
          if(allegedEventNode->Attribute("id"))
          {
             allegedEventIDRef_ = allegedEventNode->Attribute("id");
             allegedEvent_ = boost::shared_ptr<EventsChoice>(new EventsChoice(allegedEventNode));
             allegedEvent_->setID(allegedEventIDRef_);
             IDManager::instance().setID(allegedEventIDRef_,allegedEvent_);
          }
          else if(allegedEventNode->Attribute("href")) { allegedEventIDRef_ = allegedEventNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { allegedEvent_ = boost::shared_ptr<EventsChoice>(new EventsChoice(allegedEventNode));}
   }

   //partyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* partyNode = xmlNode->FirstChildElement("party");

   if(partyNode){partyIsNull_ = false;}
   else{partyIsNull_ = true;}

   if(partyNode)
   {
      for(partyNode; partyNode; partyNode = partyNode->NextSiblingElement("party")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partyNode , address : " << partyNode << std::endl;
          #endif
          if(partyNode->Attribute("href") || partyNode->Attribute("id"))
          {
              if(partyNode->Attribute("id"))
              {
                  partyIDRef_ = partyNode->Attribute("id");
                  party_.push_back(boost::shared_ptr<Party>(new Party(partyNode)));
                  party_.back()->setID(partyIDRef_);
                  IDManager::instance().setID(partyIDRef_, party_.back());
              }
              else if(partyNode->Attribute("href")) { partyIDRef_ = partyNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { party_.push_back(boost::shared_ptr<Party>(new Party(partyNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partyNode , address : " << partyNode << std::endl;
       #endif
   }

   //accountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* accountNode = xmlNode->FirstChildElement("account");

   if(accountNode){accountIsNull_ = false;}
   else{accountIsNull_ = true;}

   if(accountNode)
   {
      for(accountNode; accountNode; accountNode = accountNode->NextSiblingElement("account")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- accountNode , address : " << accountNode << std::endl;
          #endif
          if(accountNode->Attribute("href") || accountNode->Attribute("id"))
          {
              if(accountNode->Attribute("id"))
              {
                  accountIDRef_ = accountNode->Attribute("id");
                  account_.push_back(boost::shared_ptr<Account>(new Account(accountNode)));
                  account_.back()->setID(accountIDRef_);
                  IDManager::instance().setID(accountIDRef_, account_.back());
              }
              else if(accountNode->Attribute("href")) { accountIDRef_ = accountNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { account_.push_back(boost::shared_ptr<Account>(new Account(accountNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- accountNode , address : " << accountNode << std::endl;
       #endif
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<EventStatus> ConfirmationStatus::getStatus()
{
   if(!this->statusIsNull_){
        if(statusIDRef_ != ""){
             return boost::shared_static_cast<EventStatus>(IDManager::instance().getID(statusIDRef_));
        }else{
             return this->status_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<EventStatus>();
   }
}
boost::shared_ptr<EventsChoice> ConfirmationStatus::getAssertedEvent()
{
   if(!this->assertedEventIsNull_){
        if(assertedEventIDRef_ != ""){
             return boost::shared_static_cast<EventsChoice>(IDManager::instance().getID(assertedEventIDRef_));
        }else{
             return this->assertedEvent_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<EventsChoice>();
   }
}
std::vector<boost::shared_ptr<EventProposedMatch>> ConfirmationStatus::getProposedMatch()
{
   if(!this->proposedMatchIsNull_){
        if(proposedMatchIDRef_ != ""){
             return this->proposedMatch_;
        }else{
             return this->proposedMatch_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<EventProposedMatch>>();
   }
}
boost::shared_ptr<EventsChoice> ConfirmationStatus::getAllegedEvent()
{
   if(!this->allegedEventIsNull_){
        if(allegedEventIDRef_ != ""){
             return boost::shared_static_cast<EventsChoice>(IDManager::instance().getID(allegedEventIDRef_));
        }else{
             return this->allegedEvent_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<EventsChoice>();
   }
}
std::vector<boost::shared_ptr<Party>> ConfirmationStatus::getParty()
{
   if(!this->partyIsNull_){
        if(partyIDRef_ != ""){
             return this->party_;
        }else{
             return this->party_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<Party>>();
   }
}
std::vector<boost::shared_ptr<Account>> ConfirmationStatus::getAccount()
{
   if(!this->accountIsNull_){
        if(accountIDRef_ != ""){
             return this->account_;
        }else{
             return this->account_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<Account>>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-confirmation-processes-5-4/ConfirmationStatus.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ConfirmationStatus : ResponseMessage
    {
        public ConfirmationStatus(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode statusNode = xmlNode.SelectSingleNode("status");
            
            if (statusNode != null)
            {
                if (statusNode.Attributes["href"] != null || statusNode.Attributes["id"] != null) 
                {
                    if (statusNode.Attributes["id"] != null) 
                    {
                        statusIDRef_ = statusNode.Attributes["id"].Value;
                        EventStatus ob = new EventStatus(statusNode);
                        IDManager.SetID(statusIDRef_, ob);
                    }
                    else if (statusNode.Attributes["href"] != null)
                    {
                        statusIDRef_ = statusNode.Attributes["href"].Value;
                    }
                    else
                    {
                        status_ = new EventStatus(statusNode);
                    }
                }
                else
                {
                    status_ = new EventStatus(statusNode);
                }
            }
            
        
            XmlNode assertedEventNode = xmlNode.SelectSingleNode("assertedEvent");
            
            if (assertedEventNode != null)
            {
                if (assertedEventNode.Attributes["href"] != null || assertedEventNode.Attributes["id"] != null) 
                {
                    if (assertedEventNode.Attributes["id"] != null) 
                    {
                        assertedEventIDRef_ = assertedEventNode.Attributes["id"].Value;
                        EventsChoice ob = new EventsChoice(assertedEventNode);
                        IDManager.SetID(assertedEventIDRef_, ob);
                    }
                    else if (assertedEventNode.Attributes["href"] != null)
                    {
                        assertedEventIDRef_ = assertedEventNode.Attributes["href"].Value;
                    }
                    else
                    {
                        assertedEvent_ = new EventsChoice(assertedEventNode);
                    }
                }
                else
                {
                    assertedEvent_ = new EventsChoice(assertedEventNode);
                }
            }
            
        
            XmlNodeList proposedMatchNodeList = xmlNode.SelectNodes("proposedMatch");
            
            if (proposedMatchNodeList != null)
            {
                this.proposedMatch_ = new List<EventProposedMatch>();
                foreach (XmlNode item in proposedMatchNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            proposedMatchIDRef_ = item.Attributes["id"].Value;
                            proposedMatch_.Add(new EventProposedMatch(item));
                            IDManager.SetID(proposedMatchIDRef_, proposedMatch_[proposedMatch_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            proposedMatchIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        proposedMatch_.Add(new EventProposedMatch(item));
                        }
                    }
                    else
                    {
                        proposedMatch_.Add(new EventProposedMatch(item));
                    }
                }
            }
            
        
            XmlNode allegedEventNode = xmlNode.SelectSingleNode("allegedEvent");
            
            if (allegedEventNode != null)
            {
                if (allegedEventNode.Attributes["href"] != null || allegedEventNode.Attributes["id"] != null) 
                {
                    if (allegedEventNode.Attributes["id"] != null) 
                    {
                        allegedEventIDRef_ = allegedEventNode.Attributes["id"].Value;
                        EventsChoice ob = new EventsChoice(allegedEventNode);
                        IDManager.SetID(allegedEventIDRef_, ob);
                    }
                    else if (allegedEventNode.Attributes["href"] != null)
                    {
                        allegedEventIDRef_ = allegedEventNode.Attributes["href"].Value;
                    }
                    else
                    {
                        allegedEvent_ = new EventsChoice(allegedEventNode);
                    }
                }
                else
                {
                    allegedEvent_ = new EventsChoice(allegedEventNode);
                }
            }
            
        
            XmlNodeList partyNodeList = xmlNode.SelectNodes("party");
            
            if (partyNodeList != null)
            {
                this.party_ = new List<Party>();
                foreach (XmlNode item in partyNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            partyIDRef_ = item.Attributes["id"].Value;
                            party_.Add(new Party(item));
                            IDManager.SetID(partyIDRef_, party_[party_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            partyIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        party_.Add(new Party(item));
                        }
                    }
                    else
                    {
                        party_.Add(new Party(item));
                    }
                }
            }
            
        
            XmlNodeList accountNodeList = xmlNode.SelectNodes("account");
            
            if (accountNodeList != null)
            {
                this.account_ = new List<Account>();
                foreach (XmlNode item in accountNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            accountIDRef_ = item.Attributes["id"].Value;
                            account_.Add(new Account(item));
                            IDManager.SetID(accountIDRef_, account_[account_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            accountIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        account_.Add(new Account(item));
                        }
                    }
                    else
                    {
                        account_.Add(new Account(item));
                    }
                }
            }
            
        
        }
        
    
        #region Status_
        private EventStatus status_;
        public EventStatus Status_
        {
            get
            {
                if (this.status_ != null)
                {
                    return this.status_; 
                }
                else if (this.statusIDRef_ != null)
                {
                    status_ = IDManager.getID(statusIDRef_) as EventStatus;
                    return this.status_; 
                }
                else
                {
                      return this.status_; 
                }
            }
            set
            {
                if (this.status_ != value)
                {
                    this.status_ = value;
                }
            }
        }
        #endregion
        
        public string statusIDRef_ { get; set; }
        #region AssertedEvent_
        private EventsChoice assertedEvent_;
        public EventsChoice AssertedEvent_
        {
            get
            {
                if (this.assertedEvent_ != null)
                {
                    return this.assertedEvent_; 
                }
                else if (this.assertedEventIDRef_ != null)
                {
                    assertedEvent_ = IDManager.getID(assertedEventIDRef_) as EventsChoice;
                    return this.assertedEvent_; 
                }
                else
                {
                      return this.assertedEvent_; 
                }
            }
            set
            {
                if (this.assertedEvent_ != value)
                {
                    this.assertedEvent_ = value;
                }
            }
        }
        #endregion
        
        public string assertedEventIDRef_ { get; set; }
        #region ProposedMatch_
        private List<EventProposedMatch> proposedMatch_;
        public List<EventProposedMatch> ProposedMatch_
        {
            get
            {
                if (this.proposedMatch_ != null)
                {
                    return this.proposedMatch_; 
                }
                else if (this.proposedMatchIDRef_ != null)
                {
                    return this.proposedMatch_; 
                }
                else
                {
                      return this.proposedMatch_; 
                }
            }
            set
            {
                if (this.proposedMatch_ != value)
                {
                    this.proposedMatch_ = value;
                }
            }
        }
        #endregion
        
        public string proposedMatchIDRef_ { get; set; }
        #region AllegedEvent_
        private EventsChoice allegedEvent_;
        public EventsChoice AllegedEvent_
        {
            get
            {
                if (this.allegedEvent_ != null)
                {
                    return this.allegedEvent_; 
                }
                else if (this.allegedEventIDRef_ != null)
                {
                    allegedEvent_ = IDManager.getID(allegedEventIDRef_) as EventsChoice;
                    return this.allegedEvent_; 
                }
                else
                {
                      return this.allegedEvent_; 
                }
            }
            set
            {
                if (this.allegedEvent_ != value)
                {
                    this.allegedEvent_ = value;
                }
            }
        }
        #endregion
        
        public string allegedEventIDRef_ { get; set; }
        #region Party_
        private List<Party> party_;
        public List<Party> Party_
        {
            get
            {
                if (this.party_ != null)
                {
                    return this.party_; 
                }
                else if (this.partyIDRef_ != null)
                {
                    return this.party_; 
                }
                else
                {
                      return this.party_; 
                }
            }
            set
            {
                if (this.party_ != value)
                {
                    this.party_ = value;
                }
            }
        }
        #endregion
        
        public string partyIDRef_ { get; set; }
        #region Account_
        private List<Account> account_;
        public List<Account> Account_
        {
            get
            {
                if (this.account_ != null)
                {
                    return this.account_; 
                }
                else if (this.accountIDRef_ != null)
                {
                    return this.account_; 
                }
                else
                {
                      return this.account_; 
                }
            }
            set
            {
                if (this.account_ != value)
                {
                    this.account_ = value;
                }
            }
        }
        #endregion
        
        public string accountIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

